elide-async/src/main/java/com/yahoo/elide/async/export/TableExporter.java

```diff
@@ -26,44 +26,39 @@
 
 /**
  * Class for Table Export functionality.
+ * Parses a TableExport request and returns an Observable of PersistentResource.
  */
 @Slf4j
 public class TableExporter {
 
     private Elide elide;
-    private String apiVersion;
-    private User user;
-    private GraphQLParser graphQLParser;
 
-    public TableExporter(Elide elide, String apiVersion, User user) {
-        this(elide, apiVersion, user, new GraphQLParser(elide, apiVersion));
-    }
-
-    public TableExporter(Elide elide, String apiVersion, User user, GraphQLParser graphQLParser) {
+    public TableExporter(Elide elide) {
         this.elide = elide;
-        this.apiVersion = apiVersion;
-        this.user = user;
-        this.graphQLParser = graphQLParser;
     }
 
     /**
      * Exports the Data based on AsyncQuery.
      * @param query AsyncQuery object.
+     * @param user User object.
+     * @param apiVersion API Version to query.
      * @return Observable of PersistentResource.
      */
-    public Observable<PersistentResource> export(TableExport query) {
+    public Observable<PersistentResource> export(TableExport query, User user, String apiVersion) {
         Observable<PersistentResource> results = Observable.empty();
 
         UUID requestId = UUID.fromString(query.getRequestId());
 
         try (DataStoreTransaction tx = elide.getDataStore().beginTransaction()) {
             elide.getTransactionRegistry().addRunningTransaction(requestId, tx);
 
-            EntityProjection projection = null;
             RequestScope requestScope = null;
 
+            TableExportParser parser = getParser(query);
+
+            EntityProjection projection = parser.parse(query, apiVersion);
+
```

Comment:
The TableExportParser hides away the differences for different protocols (JSON-API, GraphQL, etc).  However, we still are check the queryType below to build the request scope.

We might want a different abstraction here.  Maybe the parser should return a function that runs the query - rather than a projection - so we can move together all the query type specific logic.